usmqe/web/application/implementations/web_ui.py

```
@@ -55,13 +57,21 @@ def go(self, _tries=3, *args, **kwargs):
         :return: view instance if class attribute ``VIEW`` is set or ``None``
             otherwise
         """
-        super(TendrlNavigateStep, self).go(_tries=_tries, *args, **kwargs)
+        try:
+            super(TendrlNavigateStep, self).go(_tries=_tries, *args, **kwargs)
+        except NavigationTriesExceeded:
+            now = datetime.strftime(datetime.now(), "%y_%m_%d_%H:%M")
+            self.view.browser.selenium.get_screenshot_as_file("screenshots/pre_nav" + now + ".png")
         view = self.view if self.VIEW is not None else None
         if view:
-            wait_for(
-                self.am_i_here, num_sec=10,
-                message="Waiting for view [{}] to display".format(view.__class__.__name__)
-            )
+            try:
+                wait_for(
+                    self.am_i_here, num_sec=10,
+                    message="Waiting for view [{}] to display".format(view.__class__.__name__)
+                )
+            except TimedOutError:
+                now = datetime.strftime(datetime.now(), "%y_%m_%d_%H:%M")
+                view.browser.selenium.get_screenshot_as_file("screenshots/step" + now + ".png")
```

Do we need to reraise the exception after taking a screenshot?